### PR TITLE
Tidy API and bump version

### PR DESCRIPTION
**Pull request by kw217**
Several things in here:

* Remove the error-chain implementation (and additional `to_result` method); this doesn't belong in the `-sys` crate since it's not autogenerated, and it was conflicting with the code I wanted to write in the main crate.
* Work around the DataStax CPP-502 bug with incorrectly-named symbols.
* Remove extraneous dependencies (e.g., on the log crate, which was only used in an example).
* Make the examples be proper cargo examples.
* Fix some warnings.

This has changed the API in a non-back-compatible way, so bump version to 0.10.0-pre. Update README accordingly.

### CODE REVIEW

**Comment by kw217** [comment]
@johnbatty - please can you review? Copied to @angusi for info.

File: README.md [hunk 1]
```diff
@@ -24,14 +24,16 @@ You can use this crate from cargo with
...
 ```
 
 ## Important note regarding version 0.9.0
 
-This branch is version 0.8.x of cassandra-cpp-sys. Development has begun on version 0.9.x using a newer version of bindgen,
-but it is not yet ready for use. See the [version-0.9](https://github.com/Metaswitch/cassandra-sys-rs/tree/version-0.9) branch
-to follow its development. In due course this will be merged to master, but for the moment we recommend you use version 0.8.x.
+At one point some development was done on preparing a version 0.9.0 
+using a newer version of bindgen, but the work was not completed or released. See the 
+[version-0.9](https://github.com/Metaswitch/cassandra-sys-rs/tree/version-0.9) branch
```
Comment (angusi) [inline README.md:34]: We've called this version 0.10, but this note here says there is also v0.9.
It looks like we never actually did any development on the v0.9 branch - is there actually any point in this note being here?

**Review by angusi — APPROVED** [review]
Seems pretty straightforward, and only one (non-code related) question. Thanks!

File: README.md (hunk 1 above)
Comment (kw217) [reply README.md:34]: Yes - it was pushed to crates.io as version 0.9, though as far as I can see no one ever depended on it publicly. I want to leave the door open to merging the crates again one day, so we need to treat 0.9 as reserved. This explains why, and where it went if anyone is wondering.

**Comment by kw217** [comment]
Thanks for the review!

**Pull request merged**